### PR TITLE
Add @iota/transaction package

### PR DESCRIPTION
**Pull request by laumair**
# Description
As per discussion in iotaledger#258, there is a need to package transactions related utils and validators.

This PR adds a new transaction package mostly containing existing transaction related validators and also `transactionHash` from `@iota/transaction-converter`.

**Note:** This PR doesn't remove transaction related validators from `@iota/validators` because some other packages are using these validators. Once this gets added I'll update https://github.com/iotaledger/iota.lib.js/pull/258 with the removal/deprecation of these unused validators. Will also update the dependencies for other packages using these validators. 

**Summary of changes:**
- Adds `transactionHash` that is a replica of https://github.com/iotaledger/iota.lib.js/blob/next/packages/transaction-converter/src/index.ts#L55
- Adds `isTransactionHash` that is an extended version of https://github.com/iotaledger/iota.lib.js/blob/next/packages/validators/src/validators.ts#L112
- Adds `isTransactionTrytes` that checks validity of transaction trytes. Also checks validity of transaction hash if mwm is provided
- Adds `isTransactionHashArray` that is an updated version of https://github.com/iotaledger/iota.lib.js/blob/next/packages/validators/src/validators.ts#L176. It adds an empty array check
- Adds `isTransactionTrytesArray` that is an updated version of https://github.com/iotaledger/iota.lib.js/blob/next/packages/validators/src/validators.ts#L188. It also adds an empty array check
- Adds `isAttachedTrytesArray` that is a replica of https://github.com/iotaledger/iota.lib.js/blob/next/packages/validators/src/validators.ts#L204
- Adds `isTransaction` that is an updated version of https://github.com/iotaledger/iota.lib.js/blob/next/packages/validators/src/validators.ts#L222. It adds a validation for currentIndex which makes sure that its not greater than lastIndex
- Adds `isTransactionArray` that is a replica of https://github.com/iotaledger/iota.lib.js/blob/next/packages/validators/src/validators.ts#L249
- Adds `isTailTransaction` that is a replica of https://github.com/iotaledger/iota.lib.js/blob/next/packages/validators/src/validators.ts#L358
- Adds test coverage for `isTransaction`

## Type of change

- [x] Enhancement (non-breaking change which adds functionality)

# How Has This Been Tested?

Test coverage has been added for all module methods.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/transaction/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+ * @return {boolean}
+ */
+export const isTransaction = (tx: any): tx is Transaction =>
+    isHash(tx.hash) &&
+    isTrytesOfExactLength(tx.signatureMessageFragment, SIGNATURE_MESSAGE_FRAGMENT_TRYTES_SIZE) &&
+    isHash(tx.address) &&
+    isInteger(tx.value) &&
+    isTrytesOfExactLength(tx.obsoleteTag, OBSOLETE_TAG_TRYTES_SIZE) &&
+    isInteger(tx.timestamp) &&
+    (isInteger(tx.currentIndex) && tx.currentIndex <= tx.lastIndex) &&
```
Comment (chrisdukakis) [inline packages/transaction/src/index.ts]: That's a nice idea, but I'm not sure if this check should be responsibility of bundle validator or tx validator...
If not we should also add:
```js
tx.currentIndex >= 0
```

File: packages/transaction/tslint.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (chrisdukakis) [inline packages/transaction/tslint.json]: Please remove this file, linting will move to root package.

File: packages/transaction/src/index.ts (hunk 1 above)
Comment (laumair) [reply packages/transaction/src/index.ts]: All right. Added this validation `tx.currentIndex >= 0` in `45c7515`

**Comment by laumair** [comment]
@chrisdukakis Addressed comments.

**Pull request merged**